include/boost/math/constants/constants.hpp

```diff
@@ -327,6 +327,18 @@ namespace boost{ namespace math
   BOOST_DEFINE_MATH_CONSTANT(laplace_limit, 0.662743419349181580974742097109252907056233549115022417, "0.66274341934918158097474209710925290705623354911502241752039253499097185308651127724965480259895818168")
 #endif
 
+template<typename T, typename std::enable_if<std::is_trivially_constructible<T>::value, bool>::type = true>
+constexpr T tau()
```

Comment:
Sorry to be late to the party here... I don't believe we need the two overloads here, simply:

```
template <typename T>
inline constexpr T tau() {  return two_pi<T>(); }
```

should suffice as non-trivial types can still be constexpr.

Reply:
I don't know enough to judge this, but I'm happy to change this if there is a consensus.

Reply:
> happy to change this if there is a consensus

Yes, to me also it seems like the single top-layer function should properly find the underlying overloads as suggested by @jzmaddock.

... And this has the added, subtle advantage of not creating yet another storage requirement for a separate constant.

Reply:
I pushed the change. Since @mborland kindly included a test in his diff, CI should show if it doesn't work.